Add better Error handling

julienw opened the pull request.

julienw commented:
r? @dhylands

dhylands commented on src/manager.rs:
```diff
@@ -55,19 +55,19 @@ impl Manager {
         res_to_result(unsafe {
             extern_manager::manager_add_node(self.ptr, home_id, secure)
-        })
+        }).or(Err(Error::APIError("Could not add a node")))
```
This is actually not quite right. The failure to add a node message will be in the form of a notification. The only reason that this function will fail is because an invalid home_id was passed in. And that will be similar for all of the Manager functions. The ones that return false is due to invalid parameters, not that the action actually failed.

dhylands commented on src/options.rs:
```diff
@@ -31,24 +32,30 @@ impl Options {
-        res_to_result(unsafe { extern_options::options_add_option_bool(self.ptr, name_c.as_ptr(), value) })
+        res_to_result(
+            unsafe { extern_options::options_add_option_bool(self.ptr, name_c.as_ptr(), value) }
+        ).or(Err(Error::APIError("Could not add a bool option")))
```
Similarly here. The reason for this function to fail is that the options aren't locked. So the "Could not add a bool option" message doesn't really hint at why it failed, when I think that it would be better if it did.

dhylands commented:
r=me with some rewording.

I think that "Could not add a node" would better as "AddNode: Invalid home_id" and "Could not add a bool option" should be "Could not add a bool option: not locked"

dhylands commented:
r=me with some reworded error messages.

I think "could not add a node" is probably better as something along the lines of add_node failed: invalid home_id"

And "Could not add a bool option" should be something like "Could not add a bool option: options not locked".

julienw commented:
actually, we can't add a bool option because it's _locked_ ;)
I just pushed a second commit addressing your comments, could you please have a look @dhylands ?

dhylands commented:
Looks much better.

Aside: One of my pet peeves is the "Unable to open file" error with no indication of the filename it was unable to open.
